Tools/scripts/deepfreeze.py

```diff
@@ -109,11 +109,17 @@ def __init__(self, file: TextIO) -> None:
         self.cache: Dict[tuple[type, object, str], str] = {}
         self.hits, self.misses = 0, 0
         self.patchups: list[str] = []
+        self.deallocs: list[str] = []
         self.write('#include "Python.h"')
         self.write('#include "internal/pycore_gc.h"')
         self.write('#include "internal/pycore_code.h"')
         self.write('#include "internal/pycore_long.h"')
         self.write("")
+        with self.block("static void dealloc_codeobject(PyCodeObject co)", ";"):
+            self.write("PyMem_Free(co.co_quickened);")
+            self.write("PyMem_Free(co.co_extra);")
+            with self.block("if (co.co_weakreflist != NULL)", ";"):
+                self.write("PyObject_ClearWeakRefs((PyObject *)&co);")         
```

Comment:
Also, these fields should be reset to NULL/zero, in case the interpreter is reinitialized with Py_Initialize().